doc/psidk/src/specifications/app-architecture/supervisor.md

```diff
@@ -20,48 +20,127 @@ Supervisor is responsible for:
+|   |   | [hidden] iframe.src = https://app3.psibase.tld/common/wasm-loader    |  |  |
+|   |   |----------------------------------------------------------------------|  |  |
+|   |  <-->                                                                    |  |  |
+|   |   |    wasm-loader loads plugin for the requested app                    |  |  |
+|   |   |    Plugin: https://app3.psibase.io/plugin.wasm                       |  |  |
+|   |   |                                                                      |  |  |
+|   |   '----------------------------------------------------------------------'  |  |
+|   |                                                                             |  |
+|   '-----------------------------------------------------------------------------'  |
+|                                                                                    |
+'------------------------------------------------------------------------------------'
+```
+
 The supervisor is instantiated in an iframe by a psibase app, and is responsible for instantiating plugins in their own subdomains using hidden iframes.
 
-For example, if `app.psibase` is requested from a psibase infrastructure provider, then the element stored at the root path in that service is returned to the client. That UI element is the root page of the psibase app, which will then request the supervisor from its domain on the server. After the supervisor loads, the psibase app can call into the supervisor in order to execute functionality defined in plugins.
+For example, if `app1.psibase` is requested from a psibase infrastructure provider, then the element stored at the root path in that service is returned to the client. That UI element is the root page of the psibase app, which will then request the supervisor from its domain on the server. After the supervisor loads, the psibase app can call into the supervisor in order to execute functionality defined in app2's plugin.
 
-> ⚠️ For security, plugins should only listen for messages from the supervisor. Otherwise, an attacker app could instantiate the victim app within its own iframe and impersonate the supervisor.
+> ⚠️ For security, plugins should only listen for messages from the supervisor (identified by its domain). Otherwise, an attacker app could instantiate the victim app within its own iframe and impersonate the supervisor.
 
 #### Supervisor initialization
 
-1. An app instantiates the supervisor inside a hidden iframe and gives the supervisor an ID for identifying the window/session (to help distinguish messages if the app is open in multiple browser tabs).
-2. The app also tells the supervisor the address of its own plugin.
+An app uses a library call that instantiates the supervisor inside a hidden iframe and gives the supervisor an ID for identifying the window/session (to help distinguish messages if the app is open in multiple browser tabs).
```

Comment:
The fact that we offer a library to simplify interactions with the supervisor isn't really part of the core supervisor specification.